core/common/result.hpp

```diff
@@ -103,6 +125,34 @@ namespace kagome::expected {
                             std::forward<ErrorMatch>(error_func));
     }
 
+    /**
+     * This method moves the value contained in the Result, if it was stored,
+     * or throws a NoValueException exception otherwise.
+     */
+    constexpr V tryExtractValue() {
+      return match(
+          [](Value<V> &v) -> V { return std::move(v.value); },
```

Comment:
why move?

Reply:
Because there were issues with unique_ptr stored in a Result, as it cannot be copied.  This is the solution I came up with for now. I will try to add methods which would be added to the instance only if the object is copyable or find a way to make apply_visitor return a reference, but for now this is the best solution I've found.

Reply:
since V is a generic type, maybe you can change this to be 
```
return std::forward<V>(v.value);
```
?